lib/slack-ruby-bot/commands/base.rb

```diff
@@ -122,6 +131,36 @@ def finalize_routes!
           command command_name_from_class
         end
 
+        def match_attachments(data, route)
+          fields_to_scan = %i[pretext text title]
+          data.attachments.each do |attachment|
+            fields_to_scan.each do |field|
+              next unless attachment[field]
+              match = route.match(attachment[field])
+              next unless match
+              # add access to the matched attachment
+              match.singleton_class.send(
+                :define_method, :attachment, -> { attachment }
+              )
+              match.singleton_class.send(
+                :define_method, :attachment_field, -> { field }
+              )
```

Comment:
@dblock The above code adds the access to matched attachment information to the match object instance.

Reply:
Feels way to complicated. You should just subclass the match object.

Reply:
I see but subclassing a core class without a constructor doesn't seem to be less complicate. Unless there is some easy way I missed.
I think we can use 
* a new class that uses delegation instead of inheritance
* a module and make the `match` object extend it on the fly

Reply:
I am good with a new class that uses delegation.